### PR TITLE
[BEAM-4088] Suppress a test in ExecutorServiceParallelExecutorTest

### PR DESCRIPTION
**Pull request by tgroh**
Checking to see if this stops the test from reliably failing.

DESCRIPTION HERE

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
If java doesn't turn green, I'm going to update this to suppress the test instead.

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java [hunk 1]
```diff
@@ -35,67 +35,67 @@
...
+      throw new RuntimeException(e);
     }
   }
 
   @Test
-  public void test() throws Exception {
-    pipeline
-        .apply(
-            // Use maxReadTime to force unbounded mode.
-            GenerateSequence.from(0).to(NUM_ELEMENTS).withMaxReadTime(Duration.standardDays(1)))
```
Comment (akedin) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java]: why is MaxReadTime not needed anymore?

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java [hunk 2]
```diff
@@ -35,67 +35,67 @@
...
-  @Rule
-  public final TestRule execution = outerRule(pipeline).around(threadLeakTracker);
+  @Rule public final TestRule execution = outerRule(pipeline).around(threadLeakTracker);
 
-  @Rule
-  public final TestName testName = new TestName();
+  @Rule public final TestName testName = new TestName();
 
   @Test
+  @Ignore("BEAM-4088 Test reliably fails.")
```
Comment (kennknowles) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java]: Full URL a bit nicer

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java (hunk 1 above)
Comment (tgroh) [reply runners/direct-java/src/test/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutorTest.java]: It's a bounded pipeline based on the upper bound of the `GenerateSequence`, which overrides any other way to bound it. The call doesn't do anything.

**Comment by echauchot** [comment]
Sorry @tgroh I was off, thanks for that !

**Pull request merged**